### PR TITLE
add first class support for dialogue

### PR DESCRIPTION
**Pull request by ferozco**
## Before this PR

To be able to use locally defined github YML in pdeps-bot.

## After this PR
==COMMIT_MSG==

==COMMIT_MSG==

### CODE REVIEW

File: gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjurePlugin.java [hunk 1]
```diff
@@ -169,6 +169,52 @@ private static void setupConjureJavaProject(
...
+
+    private static void setupConjureDialogueProject(
+            Project project,
+            Supplier<GeneratorOptions> optionsSupplier,
+            Task compileConjure,
+            Task compileIrTask,
+            ConjureProductDependenciesExtension productDependencyExt,
+            ExtractExecutableTask extractJavaTask) {
+        String dialogueProjectName = project.getName() + JAVA_DIALOGUE_SUFFIX;
+        if (project.findProject(dialogueProjectName) != null) {
```
Comment (iamdanfox) [inline gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjurePlugin.java]: early returns over nesting?

File: gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjurePlugin.java [hunk 2]
```diff
@@ -169,6 +169,52 @@ private static void setupConjureJavaProject(
...
+            String objectsProjectName = project.getName() + JAVA_OBJECTS_SUFFIX;
+            if (project.findProject(objectsProjectName) == null) {
+                throw new IllegalStateException(
```
Comment (iamdanfox) [inline gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjurePlugin.java]: log.warn instead maybe? if people's service definitions don't actually include _any_ conjure objects?

File: gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjurePlugin.java [hunk 3]
```diff
@@ -169,6 +169,52 @@ private static void setupConjureJavaProject(
...
+                    task.dependsOn(createWriteGitignoreTask(
+                            subproj, "gitignoreConjureDialogue", subproj.getProjectDir(), JAVA_GITIGNORE_CONTENTS));
+                    task.dependsOn(extractJavaTask);
+                });
+
+                ConjureJavaServiceDependencies.configureJavaServiceDependencies(subproj, productDependencyExt);
+                Task cleanTask = project.getTasks().findByName(TASK_CLEAN);
+                cleanTask.dependsOn(project.getTasks().findByName("cleanCompileConjureDialogue"));
+                subproj.getDependencies().add("api", project.findProject(objectsProjectName));
+                subproj.getDependencies().add("api", "com.palantir.dialogue:dialogue-target");
```
Comment (iamdanfox) [inline gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjurePlugin.java]: think the project structure is gonna be _mildly_ annoying if we want to make atomic changes across the codegen & target lirbary at the same time, as users will have to lock-step upgrade two deps or get confusnig compilation failures from generated code

**Comment by iamdanfox** [comment]
Could we hold this PR until we get a release of conjure-java that includes dialogue so we can do an actual test here?

**Review by carterkozak — APPROVED** [review]

**Pull request merged**